palette/src/yuv/yuv.rs

```diff
@@ -0,0 +1,149 @@
+    pub luminance: T,
+
+    /// The difference between overall brightness and the blue signal. This is centered around
+    /// `0.0f` but its maximum absolute value depends on the standard.
+    pub blue_diff: T,
+
+    /// The difference between overall brightness and the red signal. This is centered around
+    /// `0.0f` but its maximum absolute value depends on the standard.
+    pub red_diff: T,
+
+    /// The kind of YUV standard.
+    #[cfg_attr(feature = "serializing", serde(skip))]
+    #[palette(unsafe_zero_sized)]
+    pub standard: PhantomData<S>,
+}
+
+impl<S: YuvStandard, T: Float> Copy for Yuv<S, T> {}
+
+impl<S: YuvStandard, T: Float> Clone for Yuv<S, T> {
+    fn clone(&self) -> Yuv<S, T> {
+        *self
+    }
+}
+
+impl<S: YuvStandard, T: Float> Yuv<S, T> {
+    /// Create an YUV color (in YCbCr order)
+    pub fn new(luminance: T, blue_diff: T, red_diff: T) -> Yuv<S, T> {
+        Yuv {
+            luminance,
+            red_diff,
+            blue_diff,
+            standard: PhantomData,
+        }
+    }
+
+    fn from_rgb_internal<Sp>(rgb: Rgb<Linear<Sp>, T>) -> Self
+    where
+        T: FloatComponent,
+        Sp: RgbSpace<WhitePoint = <S::RgbSpace as RgbSpace>::WhitePoint>,
+    {
+        let xyz = Xyz::<<S::RgbSpace as RgbSpace>::WhitePoint, T>::from_color(rgb);
+        let rgb = Rgb::<(S::RgbSpace, S::TransferFn), T>::from_color(xyz);
+        let weights = S::DifferenceFn::luminance::<T>();
+        let luminance = weights[0]*rgb.red + weights[1]*rgb.green + weights[2]*rgb.blue;
+        let blue_diff = S::DifferenceFn::norm_blue(luminance - rgb.blue);
+        let red_diff = S::DifferenceFn::norm_red(luminance - rgb.red);
+
+        Yuv {
+            luminance,
+            blue_diff,
+            red_diff,
+            standard: PhantomData,
+        }
+    }
+}
+
+
+impl<S, T> Default for Yuv<S, T>
+where
+    T: Float,
+    S: YuvStandard,
+{
+    fn default() -> Yuv<S, T> {
+        Yuv::new(T::zero(), T::zero(), T::zero())
+    }
+}
+
+impl<S, T, St> From<Luma<St, T>> for Yuv<S, T>
+where
+    S: YuvStandard,
+    T: FloatComponent,
+    St: LumaStandard<WhitePoint = <S::RgbSpace as RgbSpace>::WhitePoint>,
+{
+    fn from(color: Luma<St, T>) -> Self {
+        Yuv::new(color.luma, T::zero(), T::zero())
+    }
+}
+
+impl<S, T, St> From<Rgb<St, T>> for Yuv<S, T>
+where
+    S: YuvStandard,
+    T: FloatComponent,
+    St: RgbStandard,
+    St::Space: RgbSpace<WhitePoint = <S::RgbSpace as RgbSpace>::WhitePoint>,
+{
+    fn from(color: Rgb<St, T>) -> Self {
+        let linear = color.into_linear();
+        Yuv::from_rgb_internal(linear)
+    }
+}
+
+impl<S, St, T> FromColorUnclamped<Luma<St, T>> for Yuv<S, T>
+where
+    S: YuvStandard,
+    T: FloatComponent,
+    St: LumaStandard<WhitePoint = <S::RgbSpace as RgbSpace>::WhitePoint>,
+{
+    fn from_color_unclamped(color: Luma<St, T>) -> Self {
+        let luma = color.into_linear();
+
+        Self::from(Rgb::<Linear<S::RgbSpace>, T> {
```

Comment:
This is not the same shortcut as the `From` implementation uses... :thinking:

Reply:
Whoops, I messed this up in `From<_>` assuming it was only for linear. I'll fix it and add some tests for it.